master/buildbot/www/hooks/github.py

```diff
@@ -73,7 +74,13 @@ def _get_payload(self, request):
             if hash_type != 'sha1':
                 raise ValueError('Unknown hash type: %s' % (hash_type,))
 
-            mac = hmac.new(self._secret, msg=content, digestmod=sha1)
+            _secret = self._secret
+            if isinstance(_secret, text_type):
```

Comment:
we really need an utility function for this kind of code.
One simple line has to be replaced by 6 lines here. we need something lighter.